src/elements/Checkbox/Checkbox.tsx

```diff
@@ -1,120 +1,104 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import { KeyboardEvent, MouseEvent, ReactElement, ReactNode } from "react";
-import { InputLabel } from "@elements/InputLabel/InputLabel";
 import IconCheck from "@elements/Icon/Generated/IconCheck";
 import IconMinus from "@elements/Icon/Generated/IconMinus";
-import { Style } from "@utilities/enum";
-import generateRandomId from "@utilities/generateRandomId";
+import { useCheckbox } from "@react-aria/checkbox";
+import { useFocusRing } from "@react-aria/focus";
+import { useToggleState } from "@react-stately/toggle";
+import { merge } from "@utilities/merge";
+import { FC, useRef } from "react";
 
-export enum CheckboxSelectionState {
-    Unselected = "Unselected",
-    Selected = "Selected",
-    Indeterminate = "Indeterminate",
+export enum Style {
+    Default = "Default",
+    Primary = "Primary",
 }
 
-const unselectedStyleClasses: Record<Style.Primary | Style.Secondary, string> = {
-    [Style.Primary]: "border-black-90 dark:border-white",
-    [Style.Secondary]: "border-violet-60 dark:border-violet-50",
-};
-const selectedStyleClasses: Record<Style.Primary | Style.Secondary, string> = {
-    [Style.Primary]: "bg-black-90 text-white dark:bg-white dark:text-black hover:bg-black dark:hover:bg-black-20",
-    [Style.Secondary]: "bg-violet-60 text-white dark:bg-violet-50 hover:bg-violet-70 dark:hover:bg-violet-60",
-};
-
 export type CheckboxProps = {
-    style?: Style.Primary | Style.Secondary;
-    value?: CheckboxSelectionState;
+    style?: Style;
+    checked?: "checked" | "unchecked" | "mixed";
```

Comment:
```suggestion
    checked?: CheckboxSelectionState;
```
Doesn't it makes sense to keep the interface in place?